### ISSUE TITLE
Timezone offset is added to the value after upgrading to 1.22

### ISSUE THREAD

**Issue by sisve** [briannesbitt/Carbon#863]
I'm seeing the timezone value added into the result after`addDays(1)` and `subDays(1)`. The last value ends up at 11:23:45 while still keeping the same timezone. I expect it to end up at 01:23:45.

Output from Laravel's tinker tool:
```
>>> $input = '2017-01-01T01:23:45+10:00'
=> "2017-01-01T01:23:45+10:00"
>>> $value = \Carbon\Carbon::parse($input);
=> Carbon\Carbon {#1650
     +"date": "2017-01-01 01:23:45.000000",
     +"timezone_type": 1,
     +"timezone": "+10:00",
   }
>>> $value = $value->addDays(1)
=> Carbon\Carbon {#1650
     +"date": "2017-01-02 01:23:45.000000",
     +"timezone_type": 1,
     +"timezone": "+10:00",
   }
>>> $value = $value->subDays(1)
=> Carbon\Carbon {#1650
     +"date": "2017-01-01 11:23:45.000000",
     +"timezone_type": 1,
     +"timezone": "+10:00",
   }

```

Failing test:
```
<?php

class CarbonTest extends \PHPUnit\Framework\TestCase {

    public function testCarbonZomg() {
        $input = '2017-01-01T01:23:45+10:00';

        $value = \Carbon\Carbon::parse($input);
        $this->assertEquals('2017-01-01T01:23:45+10:00', $value->toIso8601String());

        $value = $value->addDays(1);
        $this->assertEquals('2017-01-02T01:23:45+10:00', $value->toIso8601String());

        $value = $value->subDays(1);
        $this->assertEquals('2017-01-01T01:23:45+10:00', $value->toIso8601String());
        /**
         * Failed asserting that two strings are equal.
         * --- Expected
         * +++ Actual
         * @@ @@
         * -'2017-01-01T01:23:45+10:00'
         * +'2017-01-01T11:23:45+10:00'
         */
    }

}
```

**Comment by brianseitel** [comment]
I also ran into this problem, same issue. I do a lot of sensitive work with dates that, if wrong, could cost me a lot of money.

 Any chance of it getting fixed soon?

**Comment by TobiasFroberg** [comment]
This can be reproduced over here as well, running 1.22.1

**Comment by lucasmichot** [comment]
Can you check if this related to https://github.com/briannesbitt/Carbon/issues/861?

**Comment by lucasmichot** [comment]
Also have a look at https://github.com/briannesbitt/Carbon/pull/862#issuecomment-273212538

**Comment by sisve** [comment]
I'm not sure how to do that. The tinker output doesn't call toIso8601String(), I only do it to get a string for display purposes. I see no production-code in Carbon that calls toIso8601String() at all, so it isn't used by subDays()/modify().

**Comment by kaufholdr** [comment]
The problem does not exist when you use the add(CarbonInterval interval) method like:

```
$input = '2017-01-01T01:23:45+10:00'
$value = \Carbon\Carbon::parse($input);
$value = $value->add(CarbonInterval::days(1))
$value = $value->sub(CarbonInterval::days(1))
```
So I think the problem is how modify is implemented.

**Comment by Ovsyanka** [comment]
The problem exactly in "modify" implemented, i think we shoult discuss it in #883 now. There i described my vision how it should work.

**Comment by sisve** [comment]
This issue is fixed by https://github.com/briannesbitt/Carbon/pull/884

**Comment by ryanwinchester** [comment]
Just a note I also had a bunch of failing tests just using `setTimezone`

Re: https://github.com/ryanwinchester/carbonize/issues/3

**Comment by Glavic** [comment]
This issue was solved in v1.23.
